### PR TITLE
add priority for observer deletion fix #410

### LINKED ISSUE #410: [Feature]: Delete a specific observer
### Describe your use case

observer which is the combination of a pod running observer process and also the storage holding all the data, sometimes there is some problem with the storage, or the node host the observer. in these situations, add another observer and delete the abnormal one is a common procedure.

### Describe the solution you'd like

annotate the observer to tell ob-operator this observer has a higher priority to be deleted and then modify the replica of the zone this observer belongs to, ob-operator will delete the observer with the highest delete-priority.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by chris-sun-star**
## Summary

add annotation to hint priority for observer deletion
delete observer with the highest deletion priority

### CODE REVIEW

File: internal/resource/obclusteroperation/obclusteroperation_task.go [hunk 1]
```diff
@@ -157,6 +158,37 @@ func ModifyClusterSpec(m *OBClusterOperationManager) tasktypes.TaskError {
 				obcluster.Spec.Parameters = append(obcluster.Spec.Parameters, apitypes.Parameter{Name: k, Value: v})
 			}
 		}
+	case constants.ClusterOpTypeDeleteOBServers:
```
Comment (powerfooI) [inline internal/resource/obclusteroperation/obclusteroperation_task.go:161]: If user pass exact names of observers, why not delete them directly and update replicas of zones?

File: internal/const/oceanbase/misc.go [hunk 2]
```diff
@@ -13,8 +13,5 @@ See the Mulan PSL v2 for more details.
 package oceanbase
 
 const (
-	LogLevelDefault = 0
-	LogLevelInfo    = 0
-	LogLevelDebug   = 1
-	LogLevelTrace   = 2
+	DefaultDeletePriority = 100
```
Comment (powerfooI) [inline internal/const/oceanbase/misc.go:16]: Maybe `DeletionPriority` the same as other places is better.

File: internal/resource/obclusteroperation/obclusteroperation_task.go (hunk 1 above)
Comment (chris-sun-star) [reply internal/resource/obclusteroperation/obclusteroperation_task.go:161]: The problem is these are two resource managed by different controllers, if you'd like to do that way, you may add a annotation to a specific observer and modify the replica number of obzone. Adding this operation is mainly for simplicity and to use it in cli tool or dashboard.

**Review by powerfooI — APPROVED** [review]
LGTM

**Pull request merged**